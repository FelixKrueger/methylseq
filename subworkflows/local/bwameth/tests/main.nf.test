nextflow_workflow {

    name "Test Workflow BWAMETH"
    script "../main.nf"
    workflow "BWAMETH"
    config "../nextflow.config"

    setup {
        run("UNTAR") {
            script "../../../../modules/nf-core/untar/main.nf"
            process {
                """
                input[0] = [
                    [:],
                    file(params.pipelines_testdata_base_path + 'reference/Bwameth_Index.tar.gz', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("Bwameth Single-End") {

        when {
            params {
                aligner              = "bwameth"
                publish_dir_mode     = 'copy'
                outdir               = "results"
                save_reference       = false
                save_align_intermeds = false
                min_depth            = 0
                comprehensive        = false
                ignore_flags         = false
                methyl_kit           = false
                skip_deduplication   = false
            }
            workflow {
                """

                input[0] = Channel.of([
                            [ id:'test', single_end:true ],
                            file(params.pipelines_testdata_base_path + 'testdata/SRR389222_sub1.fastq.gz', checkIfExists: true)
                ])
                input[1] = UNTAR.out.untar.map { it[1] } // bwameth_index
                input[2] = Channel.of(file(params.pipelines_testdata_base_path + 'reference/genome.fa', checkIfExists: true)) // fasta
                input[3] = Channel.of(file(params.pipelines_testdata_base_path + 'reference/genome.fa.fai', checkIfExists: true)) //fasta_index
                input[4] = params.skip_deduplication
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.bam.collect { bam(it[1]).getReadsMD5() },
                    workflow.out.dedup.collect { bam(it[1]).getReadsMD5() },
                    //workflow.out.mqc,
                    workflow.out.versions
                    ).match() }
            )
        }

    }

}
