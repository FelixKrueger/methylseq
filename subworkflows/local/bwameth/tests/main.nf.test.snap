{
    "Bwameth Single-End": {
        "content": [
            [
                "37ec1c6338cc3fee7ab1cb2d48dba38"
            ],
            [
                "37ec1c6338cc3fee7ab1cb2d48dba38"
            ],
            [
                "versions.yml:md5,0906211fa5d0f6fd287c24407a882ae6",
                "versions.yml:md5,38a8061118fdcb6215a31e60419df107",
                "versions.yml:md5,3a7446087458f7f1173e10fabed1b569",
                "versions.yml:md5,3fe3921abbead6045ed1cabd18bcaece",
                "versions.yml:md5,5e8c932b160898dbcf29469f4ea6c978",
                "versions.yml:md5,68051491abbd025bd108d86911c5bf00",
                "versions.yml:md5,9b3d5f1d97768795d64f313bf20cffc0",
                "versions.yml:md5,fbbc19f0254d47d3613d2fe40ad3c926"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-09T11:03:29.575506454"
    }
}